components/server/src/repohost/repo-url.ts

```diff
@@ -10,7 +10,7 @@ export namespace RepoURL {
         repoUrl: string,
     ): { host: string; owner: string; repo: string; repoKind?: string } | undefined {
         const u = new URL(repoUrl);
-        const host = u.hostname || "";
+        const host = u.host || "";
```

Comment:
`host` contains the port, whereas `hostname` does not. 
that leads to a failure in lookup for services for host (which is taken from the SCM connection, thus needs to match)